relay-config/src/config.rs

```diff
@@ -1250,6 +1252,7 @@ impl ConfigObject for ConfigValues {
 pub struct Config {
     values: ConfigValues,
     credentials: Option<Credentials>,
+    global_config: Option<Arc<GlobalConfig>>,
```

Comment:
wasn't 100% sure if i should put it here or not. My reasoning was that if we fail to parse the file, it's better to fail early than later, and it just seems idiomatic within the codebase that we don't do these kind of risky IO-stuff in the services.

Reply:
We have a precedent where we defer the deserialization of a static config file to a service: https://github.com/getsentry/relay/blob/498401a4795a7bca5974efd0c7053b345cf0d029/relay-config/src/config.rs#L1916-L1919

That service reloads the config every 10 seconds though, which I think is overkill. In short, I think your way of doing it is better because it fails early. I'll leave final review to @jan-auer.

Reply:
I concur with this: We can load the global configs directly from within the actual service like we do it for project configs. That keeps the responsibility of the `relay-config` crate lower. The overall implementation in this PR does look good, so it would just have to be moved.

Reply:
alright, I moved it now to the spawn handler, where I'll log an error if the file exists yet it fails to load it.